### DISCUSSION TITLE
5.5 support

### DISCUSSION THREAD

**Discussion by johnburkey** [vgorloff/swift-everywhere-toolchain#131] [General]
How's 5.5?

**Comment by vgorloff** [comment]
The adoption in progress #129.
Swift 5.5 still so "fresh". Even still not available in Xcode 13 RC for macOS :0

**Comment by johnburkey** [comment]
Totally - just checking in :-)

Ping if anything weird.

**Reply by johnburkey** [reply]
watching the git, it looks like they are backporting the async stuff to older apple os'es.  Worth looking at if you run into anything weird with green thread support in droid.  Don't think you will because they did linux, just saying.
Im going to do a perf analysis of the new async stuff soon before we adopt.   And we won't adopt obviously if they don't manage to back port.

**Comment by vgorloff** [comment]
Swift 5.5 adopted in [release 1.0.74](https://github.com/vgorloff/swift-everywhere-toolchain/releases/tag/1.0.74). But, I would not recommend to use it in production. Don't even know if it works correctly. Build system warns a lot about **experimental state**.

**Comment by johnburkey** [comment]
Nice work !

I’m doing some tests on async and stuff now and then will run same on your build soon.   Over to Linux on the way.

Did you run anything on it ?

**Comment by vgorloff** [comment]
For every release I am doing a "smoke test" by running and testing this app – "https://github.com/vgorloff/swift-everywhere-samples".

**Reply by johnburkey** [reply]
Ran into a lot of issue with task groups and tasks, holding for word from Apple.  

Very unstable and not debuggable issues.   (Like tasks disappear with no except)

On top of that, when you perf test in instruments you get very bizarre stacks, and the wrong stuff in profiles. 

I filed bugs with our company's incident report quota with Apple, but no response so far for a week.    

Once this clears I'll run our stuff on Linux then droid and see where we stand.   

It's a good test, we make lots of tasks and peg the CPU at max for 100ms to do our voice recognition, and depend on task completion because we have 5-10 streaming async systems that feed on each other's outputs .

Will ping with status as it changes .

**Comment by johnburkey** [comment]
Hey, have you tested latest with code with Task & TaskGroup in it?    Porting everything to your latest , and see immediate build errors around _Concurrency and Task not being defined. (Are they in foundation now?)

<unknown>:0: warning: unable to perform implicit import of "_Concurrency" module: no such module found

CommandCell.swift:1812:17: error: cannot find 'Task' in scope

**Comment by johnburkey** [comment]
I don't see  libswift_Concurrency.so or _Concurrency.swiftmodule  in your release?   

On linux, apple platforms they are in usr/lib/swift/$OS_NAME:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx/prebuilt-modules/12.1
~/swift-5.5-RELEASE-ubuntu18.04/usr/lib/swift/linux
